access_control.py

```diff
@@ -158,16 +173,108 @@ def _process_access_control_requests(
+    )
+    revoked, failed_2 = _process_access_control_requests(
+        "revoke_access", remove=True
+    )
     return granted, revoked, failed + failed_2
 
 
+def send_pr_cirun(
+    feedstock: str,
+    feedstock_dir: str,
+    resource: str,
+    cirun_policy_args: Optional[List[str]] = None,
+) -> None:
+    """
+    Send PR to feedstock to enable Github Actions with Cirun
+
+    Parameters:
+    feedstock (str): The name of the feedstock.
+    feedstock_dir (str): Path to a git checkout of the feedstock.
+    resource (str): The name of the resource for access control.
+    cirun_policy_args (List[str]): A list of policy arguments for cirun
+                                   resources.
+    """
+
+    with update_conda_forge_config(
+        os.path.join(feedstock_dir, "recipe", "conda_build_config.yaml")
+    ) as cbc, update_conda_forge_config(
+        os.path.join(feedstock_dir, "conda-forge.yml")
+    ) as cfg:
+        if any(
+            label.startwith("cirun-")
+            for label in cbc.get("github_actions_labels", [])
+        ):
+            return
+        cfg["github_actions"] = {"self_hosted: true"}
+        if "pull_request" in cirun_policy_args:
+            cfg["github_actions"]["triggers"] = ["push", "pull_request"]
+        if "provider" not in cfg:
+            cfg["provider"] = {}
+        cfg["provider"]["linux_64"] = "github_actions"
+        if "upload_on_branch" not in cfg:
+            cfg["upload_on_branch"] = ["main"]
+        cbc["github_actions_labels"] = [resource]
+
+    gh = Github(gh_token())
+    user = gh.get_user()
+
+    repo = gh.get_repo(f"{GH_ORG}/{feedstock}-feedstock")
+    repo.create_fork()
+
+    base_branch = f"cirun-{int(time.time())}"
+
+    git_cmds = [
+        ["git", "add", "recipe/conda_build_config.yaml", "conda-forge.yml"],
+        [
+            "git",
+            "remote",
+            "add",
+            user.login,
+            "https://x-access-token:${GITHUB_TOKEN}@github.com/"
+            f"{user.login}/{feedstock}.git",
+        ],
+        [
+            "git",
+            "commit",
+            "-m",
+            f"Enable {resource} using Cirun",
+            "--author",
+            f"{user.name} <{user.email}>",
+        ],
+        ["git", "push", user.login, f"HEAD:cirun-{base_branch}"],
+    ]
+    for git_cmd in git_cmds:
+        print("Running:", *git_cmd)
+        subprocess.check_call(git_cmd)
+
+    print("Creating PR:")
+    repo.create_pull(
+        base="main",
+        head=f"{user}:{base_branch}",
+        title=f"Update feedstock to use {resource} with Cirun",
+        body=textwrap.dedent(
+            """
+        Note that only builds triggered by maintainers in the
+        who have accepted the terms of service and privacy policy will run
+        on Github actions via Cirun.
+        - [ ] Maintainers have accepted the terms of service and privacy policy
+          at https://github.com/Quansight/open-gpu-server
+        """
```

Comment:
This should only concern the resources offered by the Quansight server. We might end up using Cirun for other providers too, so maybe we should only add this message for resources with `-openstack-` in it? (Note to self: maybe we should label those with `-quansight-` too for easier identification)

Reply:
Sure, we can tweak this when we have more values for `resource`.